### PR TITLE
Vectorized Switch Node

### PR DESCRIPTION
**Pull request by kdsin23**
vectorized switch node

### CODE REVIEW

File: animation_nodes/nodes/boolean/switch_node.py [hunk 1]
```diff
@@ -27,5 +35,24 @@ def assignType(self, dataType):
...
+            else:
+                outputs.append(ifFalse)
+        if not isList(self.assignedType):
+            listDataType = toListDataType(self.assignedType)
+            return self.outputs[0].correctValue(outputs)[0]
+        else:
+            outs = []
+            for i in range(len(conditions)):
+                outs.extend(outputs[i])
+            return self.outputs[0].correctValue(outs)[0]
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/boolean/switch_node.py:58]: A virtual list can be used here. Are you familiar with those?

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]
I am not sure if this is the best way to vectorize this node. We probably need to do work around socket templates to make things easier in the future. I will get back to you on this later.

File: animation_nodes/nodes/boolean/switch_node.py (hunk 1 above)
Comment (kdsin23) [reply animation_nodes/nodes/boolean/switch_node.py:58]: Sorry, i am not familiar with virtual list, and have never used it.

Comment (OmarEmaraDev) [reply animation_nodes/nodes/boolean/switch_node.py:58]: Virtual Lists are described in https://docs.animation-nodes.com/release_notes/v2_1/#virtual-lists. Also see #1363. A lot of nodes use them, for instance, look at any of the Geometry nodes. Let me know if you need further help in understanding them.

Comment (kdsin23) [reply animation_nodes/nodes/boolean/switch_node.py:58]: Thank you very much! i will try my best to understand them.

**Pull request closed without merging**